agent_handlers.go

```diff
@@ -48,8 +50,14 @@ func (a *Agent) candidatePairRoutine() {
 }
 
 func (a *Agent) connectionStateRoutine() {
+	var wg sync.WaitGroup
 	for s := range a.chanState {
-		go a.onConnectionStateChange(s)
+		wg.Add(1)
+		go func() {
+			a.onConnectionStateChange(s)
+			wg.Done()
+		}()
+		wg.Wait()
```

Comment:
The whole sync goroutine does nothing, if this is correct just call the function.

Reply:
Thats a good point. Let me check if there arent any locks held here and we can remove the go routine.

Reply:
that didn't make a lot of sense :sweat_smile: 
Removed this. Thanks!